### PR TITLE
feat(FN-685) A record generation

### PR DESCRIPTION
**Pull request by OxonianCambion**
### Introduction

We need to be able to manage the infrastructure in declarative IaC - we've chosen Bicep.
The first phase is to be able to:

- produce a complete new deployment environment (feature) mirroring the current ones.
- be in a position to take over the extant environments.
- Note that there have been some manual changes to the environments. This work also functions as an audit of the current infrastructure.

Possible enhancements / updates will be noted but not implemented in the first phase.

This card covers:
* Wiring up A Records for all private links to enable traffic routing
### Resolution
* Adds privateDNSZoneGroups linked to the resources - this triggers the A Record generation.
NOTE: Dev has `reference-data-proxy` as well as `external-api` and a load of `demo` values. I have not attempted to recreate those.

### CODE REVIEW

File: infrastructure/resource_group_level/main.bicep [hunk 1]
```diff
@@ -304,7 +306,8 @@ module externalApi 'modules/external-api.bicep' = {
     cosmosDbDatabaseName: cosmosDbDatabaseName
     logAnalyticsWorkspaceId: logAnalyticsWorkspace.id
     acbsFunctionDefaultHostName: functionAcbs.outputs.defaultHostName
-    numberGeneratorFunctionDefaultHostName: 'TODO:FN-420'
+    numberGeneratorFunctionDefaultHostName: functionNumberGenerator.outputs.defaultHostName
```
Comment (OxonianCambion) [inline infrastructure/resource_group_level/main.bicep:309]: Drive-by fix. We have the reference available now so might as well clear it up.

File: infrastructure/resource_group_level/main.bicep [hunk 2]
```diff
@@ -357,7 +362,7 @@ module tfmApi 'modules/trade-finance-manager-api.bicep' = {
     location: location
     logAnalyticsWorkspaceId: logAnalyticsWorkspace.id
     privateEndpointsSubnetId: vnet.outputs.privateEndpointsSubnetId
-    storageAccountName: storage.outputs.storageAccountName
```
Comment (OxonianCambion) [inline infrastructure/resource_group_level/main.bicep:360]: tfm-api didn't use storageAccountName

**Review by IgnatG — APPROVED** [review]
